components/ReplyFeedback.js

```diff
@@ -70,151 +217,196 @@ const VOTE_REPLY = gql`
 `;
 
 function ReplyFeedback({
-  articleReply: {
-    articleId,
-    replyId,
-    positiveFeedbackCount,
-    negativeFeedbackCount,
-    ownVote,
-    feedbacks,
-    user,
-  } = {},
+  articleId,
+  replyId,
+  positiveFeedbackCount,
+  negativeFeedbackCount,
+  ownVote,
+  feedbacks = [],
+  reply = {},
+  className,
 }) {
-  const [downVoteDialogOpen, setDownVoteDialogOpen] = useState(false);
-  const [reasonDialogOpen, setReasonDialogOpen] = useState(false);
+  const { user, text } = reply;
+
+  const [vote, setVote] = useState(null);
+  const [reason, setReason] = useState('');
+  const [reasonsPopoverAnchorEl, setReasonsPopoverAnchorEl] = useState(null);
+  const [votePopoverAnchorEl, setVotePopoverAnchorEl] = useState(null);
+  const [tab, setTab] = useState(0);
   const [showReorderSnack, setReorderSnackShow] = useState(false);
   const currentUser = useCurrentUser();
-  const [voteReply, { loading: isVotingReply }] = useMutation(VOTE_REPLY, {
-    refetchQueries: ['LoadArticlePage'], // Update article reply order
-    awaitRefetchQueries: true,
-    onCompleted: () => setReorderSnackShow(true),
-  });
+  const [createReplyFeedback, { loading: updatingReplyFeedback }] = useMutation(
+    CREATE_REPLY_FEEDBACK,
+    {
+      refetchQueries: ['LoadArticlePage'], // Update article reply order
+      awaitRefetchQueries: true,
+      onCompleted() {
+        closeVotePopover();
+        setReason('');
+        setReorderSnackShow(true);
+      },
+    }
+  );
+
+  const openReasonsPopover = event => {
+    setReasonsPopoverAnchorEl(event.currentTarget);
+  };
+
+  const closeReasonsPopover = () => {
+    setReasonsPopoverAnchorEl(null);
+  };
+
+  const openVotePopover = (event, value) => {
+    setVotePopoverAnchorEl(event.currentTarget);
+    setVote(value);
+  };
+
+  const closeVotePopover = () => {
+    setVotePopoverAnchorEl(null);
+    setVote(null);
+  };
+
+  const classes = useStyles();
 
   // Note that currentUser and user may be undefined or null (when appId mismatch)
   // Both case should not consider as ownArticleReply
   //
   const isOwnArticleReply = currentUser && user && currentUser.id === user.id;
-  const downVoteReasons = (feedbacks || []).filter(
-    feedback => !!feedback.comment
-  );
 
   return (
-    <div className="reply-feedback">
-      {!isOwnArticleReply && <label>{t`Is the reply helpful?`}</label>}
-      <span className="vote-num">{positiveFeedbackCount}</span>
+    <div className={cx(classes.root, className)}>
       <button
-        className="btn-vote"
-        onClick={() => {
-          voteReply({ variables: { articleId, replyId, vote: 'UPVOTE' } });
-        }}
-        disabled={isOwnArticleReply || isVotingReply}
+        className={cx(classes.vote, ownVote === 'UPVOTE' && classes.voted)}
```

Comment:
Inconsistent size of thumb-up/down button and the reason button in mobile view.

![image](https://user-images.githubusercontent.com/108608/80891170-f5aebd80-8cf4-11ea-8163-818ee6de925a.png)

Corresponding mockup:
![image](https://user-images.githubusercontent.com/108608/80891197-1f67e480-8cf5-11ea-851f-39bc14f9c875.png)